lib/kafka/consumer.rb

```diff
@@ -340,12 +340,12 @@ def fetch_batches(min_bytes:, max_wait_time:)
             @logger.warn "Partition #{topic}/#{partition} is currently paused, skipping"
           else
             @logger.debug "Fetching batch from #{topic}/#{partition} starting at offset #{offset}"
-            operation.fetch_from_partition(topic, partition, offset: offset, max_bytes: max_bytes)
+            @operation.fetch_from_partition(topic, partition, offset: offset, max_bytes: max_bytes)
           end
         end
       end
 
-      operation.execute
+      @operation.execute
```

Comment:
In that case I'd probably want to create a new type of operation altogether...

Would you be okay with reviewing the current consumer code and working with me to overhaul it? I'm not completely satisfied with it, and I'd prefer to do this right. The Java client uses futures and async IO, so maybe it would be a good idea to continue down this route.